src/storage/src/table/batch_table/storage_table.rs

```diff
@@ -796,10 +796,10 @@ impl<S: StateStore, SD: ValueRowSerde> StorageTableInner<S, SD> {
         let iter = match iterators.len() {
             0 => unreachable!(),
             1 => iterators.into_iter().next().unwrap(),
-            // Concat all iterators if not to preserve order.
-            _ => futures::stream::iter(iterators.into_iter().map(Box::pin).collect_vec())
-                .flatten_unordered(1024),
-        };
+            // All need merge
+            _ => merge_sort(iterators.into_iter().map(Box::pin).collect()),
```

Comment:
Like the normal batch iterator, can we have a `ordered` option to determine whether `merge_sort` is needed?

See https://github.com/risingwavelabs/risingwave/blob/d8199b5eeb44368109e123a9e4b586bcca1aac48/src/batch/src/executor/row_seq_scan.rs#L238 and https://github.com/risingwavelabs/risingwave/blob/d8199b5eeb44368109e123a9e4b586bcca1aac48/src/storage/src/table/batch_table/storage_table.rs#L539